scripts/engine/component.py

```diff
@@ -144,20 +145,56 @@ class Knowledge(RegisteredComponent):
     An entity's knowledge, including skills. Skills are held as skill_name : {Skill, cooldown}.
     """
 
-    def __init__(self, skills: Dict[str, Dict[str, Any]] = None, skill_order: List[str] = None):
-        if skills is None:
-            skills = {}
-        if skill_order is None:
-            skill_order = []
-
-        self.skill_order = skill_order  # list of skill names, to allow access by index
-        self.skills: Dict[str, Dict[str, Any]] = skills  # skill_name : {Skill, cooldown}
-            # FIXME - how is it str and Skill? Can't be both
+    def __init__(self, skills: List[Skill] = None, skill_order: List[str] = None):
+        skills = skills or {}
+        skill_order = skill_order or []
```

Comment:
```
if skills_order == None:
    skills_order = []
```
Is the same as
```
skills_order = skills_order or []
```
Why? Basically if `skills_order` is `None` it evaluates to false and we get the second part of the conditional. I felt it was cleaner and less cluttery (more pythonic too and uses less lines!)

By the way, this is called the [null coalescing operator](https://en.wikipedia.org/wiki/Null_coalescing_operator), I usually find it pretty useful.